### ISSUE TITLE
require options

### ISSUE THREAD

**Issue by digitalbanana** [jmcdo29/nest-commander#37]
Hi there,

Love the nest-commander package - very helpful to get cli functionality with nestjs :-) 

Have hit a slight issue where adding the hint to indicate that a flag is required doest seem to be honoured, is that right or am I using it wrong ly. Example below to hopefully clarify what I mean...

so if i want to make an arg `--sample-arg` required I should

so according to the [docs](https://github.com/jmcdo29/nest-commander/blob/main/packages/nest-commander/README.md#option), using `<>` should make the flag required 
```
  @Option({
    flags: '-s --sample-arg <sampleArg>',
  })
  parseArg(val: string): string {
    return val;
  }
```
however i can execute the command above without supplying `-s` flag so it looks like the option below (using `[]` syntax) is effectively the same
```
  @Option({
    flags: '-s --sample-arg [sampleArg]',
  })
  parseArg(val: string): string {
    return val;
  }
```

**Comment by jmcdo29** [comment]
Do you have a minimum reproduction of this?

**Comment by digitalbanana** [comment]
yep hopefully exemplified here https://github.com/digitalbanana/nest-commander-exemplar

**Comment by jmcdo29** [comment]
Thanks for the reproduction. 

With commander and options, the `[]` or `<>` syntax determines if a value is required if the option is passed. So if you do `node dist/main basic -s` it's fine because `-s, --string [string]` is used, and you'll get back `true`, but if you do `node dist/main basic -n` you'll get an error at runtime because the value for `number` is missing.

Commander has the idea of [`requiredOptions`](https://www.npmjs.com/package/commander#required-option) which are options that are required to be run with the command, i.e. `node dist/main basic` would then throw an error because it's missing the `-n` option. I can work on adding that in via an option on the decorator, as right now options are only added in with `program.option` and not `program.requiredOption`.

**Comment by digitalbanana** [comment]
ah ok that makes sense - thanks for clarifying....

required option would be super useful, am happy to contribute if want any help but need a pointer of where to begin :-p

**Comment by jmcdo29** [comment]
We'll need to update the [`OptionMetada`](https://github.com/jmcdo29/nest-commander/blob/main/packages/nest-commander/src/command-runner.interface.ts#L37) to include something like `requiredOption` as a boolean. And probably a bit of refactoring would be necessary on [this method of the command-runner service class](https://github.com/jmcdo29/nest-commander/blob/main/packages/nest-commander/src/command-runner.service.ts#L34) so that we can track this `requiredOption` value. Then we should just need to make a switch [here](https://github.com/jmcdo29/nest-commander/blob/main/packages/nest-commander/src/command-runner.service.ts#L58) to use `requiredOption` instead of `option`. Feel free to take a shot at it if you feel comfortable with it

**Comment by SuEric** [comment]
nice feature request! 

@jmcdo29 I am giving a shot today to your cli vs `nestjs-console`. 

This console looks very promising! Best of luck

**Comment by jmcdo29** [comment]
Let me know what you think. Also, I'm active on [Nest's Discord Server](https://discord.gg/nestjs) where there's a dedicated channel for #nest-commander so feel free to message there too.

**Comment by jmcdo29** [comment]
@digitalbanana just checking in to see how progress on this is coming if you've had the time. No worries if not

**Comment by jmcdo29** [comment]
🏓

**Comment by digitalbanana** [comment]
sorry @jmcdo29 went on holiday and was only doing minimal work stuff - looking at this now 👍

**Comment by jmcdo29** [comment]
This has been added and published in `nest-commander@2.1.0`

**Comment by digitalbanana** [comment]
awesome thanks @jmcdo29
